store/types/pruning.go

```diff
@@ -2,18 +2,45 @@ package types
 
 // PruningStrategy specifies how old states will be deleted over time where
 // keepRecent can be used with keepEvery to create a pruning "strategy".
+// CONTRACT: snapshotEvery is a multiple of keepEvery
```

Comment:
Can you explain the difference between keepEvery and snapshotEvery here?
This is important, yet neither are defined above (actually keepRecent is not really defined here either)

Reply:
@AdityaSripal I thought we agreed to remove `keepRecent` in that it should never be not 1, so no sense in having a user provide a value.

```go
// PruningOptions defines the specific pruning strategy every store in a multi-store will use
// when committing state, where keepEvery determines which committed heights are flushed
// to disk and snapshotEvery determines which of these heights are kept after pruning. Note,
// that the invariant keepEvery % snapshotEvery = 0 must hold. 
type PruningOptions struct {
  keepEvery     int64
  snapshotEvery int64
}
```

EDIT: I've realized that maybe we do need `keepRecent` for when we want to prune *everything*. But is this the only case? We should clearly define when `keepRecent` is allowed not to be 1 and what it means for it to be not 1.